src/templates/footer-block/index.vue

```diff
@@ -101,61 +109,71 @@
   </footer>
 </template>
 
-<script>
+<script setup>
+  import { ref, computed } from 'vue'
   import { useAccountStore } from '@/stores/account'
   import SystemStatusBarBlock from '@templates/system-status-bar-block'
-  import { mapActions, mapState } from 'pinia'
   import PrimeButton from 'primevue/button'
   import Dropdown from 'primevue/dropdown'
+  import { useRoute } from 'vue-router'
+  const accountStore = useAccountStore()
+  const setTheme = accountStore.setTheme
+  const route = useRoute()
+
+  defineOptions({
+    name: 'footer-block'
+  })
+
+  const themeOptions = ref([
+    { name: 'Light', value: 'light', icon: 'pi pi-sun' },
+    { name: 'Dark', value: 'dark', icon: 'pi pi-moon' },
+    { name: 'System', value: 'system', icon: 'pi pi-desktop' }
+  ])
+
+  const currentTheme = computed(() => accountStore.currentTheme)
+  const selectedTheme = computed(() =>
+    themeOptions.value.find((option) => option.value === currentTheme.value)
+  )
+
+  const classObjectRoot = computed(() => ({
+    'flex max-md:flex-col': route.meta.hideLinksFooter,
+    'grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:lg:grid-cols-3': !route.meta.hideLinksFooter
+  }))
+
+  const classObjectOptionButton = computed(() => ({
+    'justify-center md:justify-start xl:justify-end': !route.meta.hideLinksFooter,
+    'md:justify-end justify-center': route.meta.hideLinksFooter
+  }))
```

Comment:
My suggestion is to return and object based on the route meta value:
const dynamicClassesByRouteMeta = computed ()=>({
footer: route.meta.hideLinksFooter? 'flex max-md:flex-col': 'grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:lg:grid-cols-3',
optionButton: route.meta.hideLinksFooter ? 'md:justify-end justify-center' : 'justify-center md:justify-start xl:justify-end'
})

and use this object on the template:
dynamicClassesByRouteMeta.footer and dynamicClassesByRouteMeta.optionButton